src/main/java/com/iexec/worker/pubsub/StompClientService.java

```diff
@@ -91,129 +76,41 @@ Optional<Subscription> subscribeToTopic(String topic, StompFrameHandler messageH
-                backOff();
-                // Reset the latch to mark requests as treated.
-                sessionRequest = new CountDownLatch(1);
-                // Send one request to the server.
-                createSession();
-            } catch(InterruptedException e) {
-                log.error("STOMP session request listener got interrupted", e);
-                // Release semaphore so another thread can be started.
-                listenerLock.release();
-                // Properly handle InterruptedException.
-                Thread.currentThread().interrupt();
-                // The thread will stop
-            } catch(Exception e) {
-                // Ignore all errors and continue
-                log.error("An error occurred while listening to STOMP session requests", e);
-                // The thread will continue
-            }
-        }
+    @Scheduled(fixedRate = SESSION_REFRESH_DELAY_MS)
+    void scheduleStompSessionCreation() {
+        AsyncUtils.runAsyncTask("listen-to-stomp-session",
+                this::createStompSessionIfDisconnected, singleThreadExecutor);
     }
 
     /**
-     * Checks if a new STOMP session must be created.
-     * <p>
-     * If a session exists and is connected, nothing is done.
-     * Otherwise, the latch is decremented and reaches {@literal 0}.
-     * This unblocks {@code sessionRequest.await()}.
+     * Create a new STOMP session
+     * if none exists or the last one is disconnected.
+     * @return A {@link String} representing the session ID.
      */
-    void requestNewSession() {
+    String createStompSessionIfDisconnected() {
         if (stompSession != null && stompSession.isConnected()) {
-            log.info("A valid STOMP session exists, ignoring this request");
-            return;
+            log.debug("A valid STOMP session exists, ignoring this request");
+            return stompSession.getSessionId();
         }
-        sessionRequest.countDown();
-        log.info("Requested a new STOMP session");
-    }
 
-    /**
-     * Establish connection to the WebSocket Server.
-     */
-    void createSession() {
-        log.info("Creating new STOMP session");
-        stompClient.connect(webSocketServerUrl, new SessionHandler());
-    }
-
-    /**
-     * Sleep {@link StompClientService#SESSION_REFRESH_BACK_OFF_DELAY} seconds.
-     * <p>
-     * If no STOMP session exists, we are creating the first session and there is no need to wait.
-     * @throws InterruptedException if thread is interrupted.
-     */
-    void backOff() throws InterruptedException {
-        if (stompSession != null) {
-            TimeUnit.SECONDS.sleep(SESSION_REFRESH_BACK_OFF_DELAY);
+        try {
+            log.info("Creating new STOMP session");
+            this.stompSession = stompClient
+                    .connect(webSocketServerUrl, new SessionHandler())
+                    .get(SESSION_REFRESH_DELAY_MS, TimeUnit.MILLISECONDS);
```

Comment:
Isn't 5 seconds a little short for a timeout? I think something like 30s is closer to a standard

Reply:
This may be short, right. As per a standard, there doesn't seem to exist any. I find some 15 seconds, 1 minute, 2 minutes, ... Every case seems to have its own timeout. If you agree, we can set a 15 seconds timeout. I am quite reluctant to set a too high timeout to avoid being stuck in this thread.

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/492/commits/ce7a9905b94441a6e87d0c3b298b9859bf88a116.